src/client/components/Inputs/Select/Indicators/CollapsibleGroup/hooks/useExpandGroup.ts

```diff
@@ -0,0 +1,38 @@
+import { useCallback, useEffect, useState } from 'react'
+
+type Props = {
+  inputValue: string
+}
+
+type Returned = {
+  expanded: boolean
+  toggleExpanded: () => void
+}
+
+export const useExpandGroup = (props: Props): Returned => {
+  const { inputValue } = props
+  const [expanded, setExpanded] = useState(false)
+  const [manuallyExpanded, setManuallyExpanded] = useState(false)
+
+  useEffect(() => {
+    if (inputValue.length > 0) {
+      // Expand automatically on search
+      if (!expanded) {
+        setExpanded(true)
+        setManuallyExpanded(false)
+      }
+    } else if (!manuallyExpanded) {
+      // Collapse if not manually expanded
+      if (expanded) {
+        setExpanded(false)
+      }
+    }
+  }, [expanded, inputValue, manuallyExpanded])
```

Comment:
One caveat here is that when you manually open a region but the search string you type doesn't match any of its countries, react-select will internally remove this Group component (and its state). So, when the search is cleared and the component is rendered again, it has the default initial state. 

example:

https://github.com/user-attachments/assets/b5e4134a-b51e-4ef7-a56e-6ab43f1afef3

Unfortunately react-select doesn't have a prop or custom option to prevent this. One possible solution could be to add a Context provider to the CountrySelect and store the `expanded` state for each Group using a Group id.